File: simple_geo3x3.pl
#!/usr/bin/perl
 
use lib qw(./);
use geo3x3;

my $code = geo3x3::encode(35.65858, 139.745433, 14);
print $code . "\n";

my ($lat, $lng, $level, $unit) = geo3x3::decode("E3793653391822");
print $lat . " " . $lng . " " . $level . " " . $unit . "\n";

File: simple_geo3x3.cpp
#include <iostream>
#include "geo3x3.h"
using namespace std;

int main() {
    enum GEO3X3_RES err;

    char enc_buf[16];
    struct geo3x3_wgs84 res;

    if ((err = geo3x3_from_wgs84_str(
        35.65858, 139.745433,
        14,
        enc_buf,
        sizeof(enc_buf)
    ))) {
        // handle errors
        exit(1);
    }

    cout << enc_buf << endl;

    if ((err = geo3x3_to_wgs84_str(enc_buf, &res))) {
        // handle errors
        exit(1);
    }

    cout << res.lat << " " << res.lng << " " << +res.level << " " << res.unit << endl;

    return 0;
}
